kedro/contrib/io/matplotlib/matplotlib_writer.py

```diff
@@ -0,0 +1,139 @@
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+``MatplotlibWriter ``  saves matplotlib objects as image files.
+"""
+
+import os.path
+from typing import Any, Dict, Optional
+
+from kedro.io import AbstractDataSet, DataSetError
+
+
+class MatplotlibWriter(AbstractDataSet):
+    """
+        ``MatplotlibWriter`` saves matplotlib objects as image files.
+
+        Example:
+        ::
+
+            >>> import matplotlib.pyplot as plt
+            >>> from kedro.contrib.io.matplotlib import MatplotlibWriter
+            >>>
+            >>> plt.plot([1,2,3],[4,5,6])
+            >>>
+            >>> single_plot_writer = MatplotlibWriter(filepath="docs/new_plot.png")
+            >>> single_plot_writer.save(plt)
+            >>>
+            >>> plt.close()
+            >>>
+            >>> plots = dict()
+            >>>
+            >>> for colour in ['blue', 'green', 'red']:
+            >>>     plots[colour] = plt.figure()
+            >>>     plt.plot([1,2,3],[4,5,6], color=colour)
+            >>>     plt.close()
+            >>>
+            >>> multi_plot_writer = MatplotlibWriter(filepath="docs/",
+            >>>                                      save_args={'multiFile': True})
+            >>> multi_plot_writer.save(plots)
+
+    """
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Optional[Dict[str, Any]] = None,
+        save_args: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            filepath: path to a text file.
+            load_args: Currently ignored as loading is not supported.
+            save_args: multiFile: allows for multiple plot objects
+                to be saved. Additional load arguments can be found at
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+        default_save_args = {"multiFile": False}
+        default_load_args = {}
+
+        self._filepath = filepath
+        self._load_args = self._handle_default_args(load_args, default_load_args)
+        self._save_args = self._handle_default_args(save_args, default_save_args)
+        self._mutlifile_mode = self._save_args.pop("multiFile")
+
+    @staticmethod
+    def _handle_default_args(user_args: dict, default_args: dict) -> dict:
```

Comment:
Can we make this class extend `DefaultArgsMixin` instead of having this static method, ensuring `super().__init__()` is called at the top? That will take care of save_args & default_save_args, etc., and is consistent with other datasets in contrib.